lib/Dialect/EmitC/Conversion/TosaToEmitC.cpp

```diff
@@ -223,6 +223,48 @@ class MatMulOpConversion : public OpConversionPattern<tosa::MatMulOp> {
   }
 };
 
+class ReluNOpConversion : public OpConversionPattern<tosa::ReluNOp> {
+  using OpConversionPattern<tosa::ReluNOp>::OpConversionPattern;
+
+public:
+  ReluNOpConversion(MLIRContext *ctx)
+      : OpConversionPattern<tosa::ReluNOp>(ctx) {}
+
+private:
+  LogicalResult
+  matchAndRewrite(tosa::ReluNOp reluNOp, ArrayRef<Value> operands,
+                  ConversionPatternRewriter &rewriter) const override {
+
+    StringRef funcName = "tosa::reluN";
+    StringAttr callee = rewriter.getStringAttr(funcName);
+
+    SmallVector<Attribute, 2> args_;
+    args_.push_back(rewriter.getIndexAttr(0));
+
+    // Determine to which max value we have to clamp to
+    auto operandType =
+        operands[0].getType().cast<RankedTensorType>().getElementType();
+    if (operandType.isSignedInteger() || operandType.isSignlessInteger() ||
+        operandType.isUnsignedInteger()) {
+      args_.push_back(reluNOp.max_intAttr());
+    } else if (operandType.isF16() || operandType.isF32() ||
+               operandType.isF64() || operandType.isF80() ||
+               operandType.isF128()) {
```

Comment:
I think you can use `isa<FloatType>(operandType)` and `isa<IntegerType>(operandType)`
And maybe call the variable `elementType`.

Reply:
@simon-camp I was looking for something like that but couldnt find it! Thanks!

Reply:
Sometimes it's worth to take a look into the printer :)
https://github.com/iml130/mlir-emitc/blob/8f4f56c2d5042dd4e770a1ac9416f989e0480662/lib/Target/Cpp/TranslateToCpp.cpp#L37-L38